src/utils/useKeyboardBottomInset.ts

```diff
@@ -9,7 +9,7 @@ export const useKeyboardBottomInset = () => {
     function onKeyboardChange(e: KeyboardEvent) {
       if (
         e.startCoordinates &&
-        e.endCoordinates.screenY < e.startCoordinates.screenY
+        e.endCoordinates.screenY <= e.startCoordinates.screenY
```

Comment:
../node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java
<img width="548" alt="image" src="https://user-images.githubusercontent.com/14011323/232781686-e1400325-15a7-46c5-a595-f06cf7651205.png">
This is the createKeyboardEventPayload source code from react native. It does not contain the startCoordinates field. So that is avoidKeyboard the reason for not working on the Android platform.